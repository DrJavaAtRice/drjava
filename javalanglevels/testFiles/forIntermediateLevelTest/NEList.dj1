public class NEList implements IList2 {

   Object _first;
   IList2 _rest;

   
   public NEList() {
     _first = new Integer(1);
     _rest = MTList.Singleton;
   }
   
   public int sum() {

     return ((Integer)_first) + _rest.sum();

   }

   public String toString2() {

    return "(" + _first + "," + _rest.toStringHelp() + ")";

   }

   

   public String toStringHelp() {

     return " ," + _first + _rest.toStringHelp();

   }

   public int getlength(){

    return _rest.getlengthHelp(1); 

}

   public int getlengthHelp(int acc){

     return _rest.getlengthHelp(acc + 1);}

}