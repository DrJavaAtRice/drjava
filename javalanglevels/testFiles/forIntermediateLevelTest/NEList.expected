// Language Level Converter line number map: dj*->java. Entries: 42
//     1->8         2->9         3->10        4->11        5->12        6->13        7->14        8->15   
//     9->16       10->17       11->18       12->19       13->20       14->21       15->22       16->23   
//    17->24       18->25       19->26       20->27       21->28       22->29       23->30       24->31   
//    25->32       26->33       27->34       28->35       29->36       30->37       31->38       32->39   
//    33->40       34->41       35->42       36->43       37->44       38->45       39->46       40->47   
//    41->48       42->49   
public class NEList implements IList2 {

   private final Object _first;
   private final IList2 _rest;

   
   public NEList() {
     _first = new Integer(1);
     _rest = MTList.Singleton;
   }
   
   public int sum() {

     return ((Integer)_first) + _rest.sum();

   }

   public String toString2() {

    return "(" + _first + "," + _rest.toStringHelp() + ")";

   }

   

   public String toStringHelp() {

     return " ," + _first + _rest.toStringHelp();

   }

   public int getlength(){

    return _rest.getlengthHelp(1); 

}

   public int getlengthHelp(final int acc){

     return _rest.getlengthHelp(acc + 1);}


  /** This method is automatically generated by the Language Level Converter. */
  public NEList(java.lang.Object _first, IList2 _rest) {
    super();
    this._first = _first;
    this._rest = _rest;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.Object _first() {
    return _first;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public IList2 _rest() {
    return _rest;
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        _first() + ", " + 
        _rest() + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      NEList cast = ((NEList) o);
      return (_first() != null && _first().equals(cast._first())) && 
          (_rest() != null && _rest().equals(cast._rest()));
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode() ^ 
        (_first() == null ? 0 : _first().hashCode()) ^ 
        (_rest() == null ? 0 : _rest().hashCode());
  }
}
